near-token-factory/src/lib.rs

```diff
@@ -43,39 +56,68 @@ pub struct Contract {
 }
 
 // TODO: Add pausable
-// TODO: Add access control
 #[near_bindgen]
 impl Contract {
     /// Initializes the contract. The locker account id MUST be the NEAR
     /// representative of the Aurora address of the locker contract created
     /// using the Cross Contract Call interface.
     #[init]
-    pub fn new(aurora: AccountId, locker: aurora_sdk::Address) -> Self {
+    pub fn new(
+        aurora: AccountId,
+        locker: aurora_sdk::Address,
+        super_admin: Option<AccountId>,
+    ) -> Self {
         require!(
             env::current_account_id().as_str().len() + 1 + 40 <= 63,
             ERR_INVALID_ACCOUNT
         );
 
-        Self {
+        // If not specified, the super_admin is the deployer of this contract.
+        let super_admin = super_admin.unwrap_or_else(env::predecessor_account_id);
+
+        let mut contract = Self {
             aurora,
+            token_super_admin: super_admin.clone(),
             token_binary: LazyOption::new(StorageKey::TokenBinary, None),
             token_binary_version: 0,
             tokens: UnorderedMap::new(StorageKey::TokenMap),
             locker,
-        }
+            __acl: Default::default(),
+        };
+
+        // Initialize Acl permissions.
+        require!(
+            contract.acl_init_super_admin(super_admin.clone()),
+            "Failed to add factory as initial acl super-admin",
+        );
+
+        require!(contract
+            .acl_grant_role(AclRole::TokenBinaryUpdater.into(), super_admin.clone())
+            .unwrap_or_default());
```

Comment:
```suggestion
        require!(
            contract
                .__acl
                .grant_role_unchecked(AclRole::TokenBinaryUpdater, &super_admin),
                "Failed to grant AclRole::TokenBinaryUpdater",
        );
```

This call of `acl_grant_role` succeeds only if the contract itself is super admin or admin for `TokenBinaryUpdater`. Neither of it holds if input parameter `super_admin` is `Some(other_account_id)`.